In [4]:
cd C:/Users/Watson/Desktop

C:\Users\Watson\Desktop


In [5]:
#Import Libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

In [7]:
od = pd.read_csv("CT_Drug_Deaths_2012-2018.csv")

In [8]:
od.head()

ID            Date      DateType   Age     Sex          Race  \
0  14-0273  6/28/2014 0:00  DateReported   NaN     NaN           NaN   
1  13-0102  3/21/2013 0:00   DateofDeath  48.0    Male         Black   
2  16-0165  3/13/2016 0:00   DateofDeath  30.0  Female         White   
3  16-0208  3/31/2016 0:00   DateofDeath  23.0    Male         White   
4  13-0052  2/13/2013 0:00   DateofDeath  22.0    Male  Asian, Other   

  ResidenceCity ResidenceCounty ResidenceState  DeathCity  \
0           NaN             NaN            NaN        NaN   
1       NORWALK             NaN            NaN    NORWALK   
2    SANDY HOOK       FAIRFIELD             CT    DANBURY   
3           RYE     WESTCHESTER             NY  GREENWICH   
4      FLUSHING          QUEENS            NaN  GREENWICH   

              ...              Tramad Morphine_NotHeroin Hydromorphone Other  \
0             ...                 NaN                NaN           NaN   NaN   
1             ...                 NaN                NaN           NaN   NaN   
2             ...                 NaN                NaN           NaN   NaN   
3             ...                 NaN                NaN           NaN   NaN   
4             ...                 NaN                NaN           NaN   NaN   

  OpiateNOS AnyOpioid MannerofDeath                            DeathCityGeo  \
0       NaN       NaN      Accident             CT\n(41.575155, -72.738288)   
1       NaN       NaN      Accident     Norwalk, CT\n(41.11805, -73.412906)   
2       NaN         Y      Accident    Danbury, CT\n(41.393666, -73.451539)   
3       NaN         Y      Accident  Greenwich, CT\n(41.026526, -73.628549)   
4       NaN       NaN      Accident  Greenwich, CT\n(41.026526, -73.628549)   

                          ResidenceCityGeo                InjuryCityGeo  
0              CT\n(41.575155, -72.738288)  CT\n(41.575155, -72.738288)  
1      NORWALK, CT\n(41.11805, -73.412906)  CT\n(41.575155, -72.738288)  
2  SANDY HOOK, CT\n(41.419998, -73.282501)                          NaN  
3                                      NaN                          NaN  
4                                      NaN  CT\n(41.575155, -72.738288)  

[5 rows x 41 columns]

In [9]:
od.dtypes

ID                      object
Date                    object
DateType                object
Age                    float64
Sex                     object
Race                    object
ResidenceCity           object
ResidenceCounty         object
ResidenceState          object
DeathCity               object
DeathCounty             object
Location                object
LocationifOther         object
DescriptionofInjury     object
InjuryPlace             object
InjuryCity              object
InjuryCounty            object
InjuryState             object
COD                     object
OtherSignifican         object
Heroin                  object
Cocaine                 object
Fentanyl                object
FentanylAnalogue        object
Oxycodone               object
Oxymorphone             object
Ethanol                 object
Hydrocodone             object
Benzodiazepine          object
Methadone               object
Amphet                  object
Tramad                  object
Morphine

In [11]:
# see if any columns have nulls
display(od.isnull().any())

ID                     False
Date                    True
DateType                True
Age                     True
Sex                     True
Race                    True
ResidenceCity           True
ResidenceCounty         True
ResidenceState          True
DeathCity               True
DeathCounty             True
Location                True
LocationifOther         True
DescriptionofInjury     True
InjuryPlace             True
InjuryCity              True
InjuryCounty            True
InjuryState             True
COD                    False
OtherSignifican         True
Heroin                  True
Cocaine                 True
Fentanyl                True
FentanylAnalogue        True
Oxycodone               True
Oxymorphone             True
Ethanol                 True
Hydrocodone             True
Benzodiazepine          True
Methadone               True
Amphet                  True
Tramad                  True
Morphine_NotHeroin      True
Hydromorphone           True
Other         

In [12]:
od.shape

(5105, 41)

In [13]:
od.describe(include='all')

ID            Date     DateType          Age   Sex   Race  \
count      5105            5103         5103  5102.000000  5099   5092   
unique     5105            2098            2          NaN     3     11   
top     16-0343  8/18/2017 0:00  DateofDeath          NaN  Male  White   
freq          1               9         2822          NaN  3773   4004   
mean        NaN             NaN          NaN    41.964916   NaN    NaN   
std         NaN             NaN          NaN    12.336639   NaN    NaN   
min         NaN             NaN          NaN    14.000000   NaN    NaN   
25%         NaN             NaN          NaN    32.000000   NaN    NaN   
50%         NaN             NaN          NaN    42.000000   NaN    NaN   
75%         NaN             NaN          NaN    52.000000   NaN    NaN   
max         NaN             NaN          NaN    87.000000   NaN    NaN   

       ResidenceCity ResidenceCounty ResidenceState DeathCity  \
count           4932            4308           3556      5100   
unique           370              84             26       222   
top         HARTFORD        HARTFORD             CT  HARTFORD   
freq             296            1205           3442       563   
mean             NaN             NaN            NaN       NaN   
std              NaN             NaN            NaN       NaN   
min              NaN             NaN            NaN       NaN   
25%              NaN             NaN            NaN       NaN   
50%              NaN             NaN            NaN       NaN   
75%              NaN             NaN            NaN       NaN   
max              NaN             NaN            NaN       NaN   

                   ...              Tramad Morphine_NotHeroin Hydromorphone  \
count              ...                 130                 42            25   
unique             ...                   1                  5             1   
top                ...                   Y                  Y             Y   
freq               ...                 130                 38            25   
mean               ...                 NaN                NaN           NaN   
std                ...                 NaN                NaN           NaN   
min                ...                 NaN                NaN           NaN   
25%                ...                 NaN                NaN           NaN   
50%                ...                 NaN                NaN           NaN   
75%                ...                 NaN                NaN           NaN   
max                ...                 NaN                NaN           NaN   

       Other OpiateNOS AnyOpioid MannerofDeath  \
count    435        88      2466          5095   
unique   134         1         2             5   
top      PCP         Y         Y      Accident   
freq      48        88      2460          5066   
mean     NaN       NaN       NaN           NaN   
std      NaN       NaN       NaN           NaN   
min      NaN       NaN       NaN           NaN   
25%      NaN       NaN       NaN           NaN   
50%      NaN       NaN       NaN           NaN   
75%      NaN       NaN       NaN           NaN   
max      NaN       NaN       NaN           NaN   

                                 DeathCityGeo  \
count                                    5105   
unique                                    374   
top     Hartford, CT\n(41.765775, -72.673356)   
freq                                      379   
mean                                      NaN   
std                                       NaN   
min                                       NaN   
25%                                       NaN   
50%                                       NaN   
75%                                       NaN   
max                                       NaN   

                             ResidenceCityGeo                InjuryCityGeo  
count                                    5012                         5027  
unique                                    289     

5105 deaths but only 5095 manner of death determined. We'll start here for dropping nulls

In [17]:
od = od.dropna(subset=['MannerofDeath'])

In [18]:
od.describe(include='all')

ID            Date     DateType          Age   Sex   Race  \
count      5095            5094         5094  5093.000000  5090   5083   
unique     5095            2096            2          NaN     3     11   
top     16-0343  8/18/2017 0:00  DateofDeath          NaN  Male  White   
freq          1               9         2816          NaN  3764   3998   
mean        NaN             NaN          NaN    41.950913   NaN    NaN   
std         NaN             NaN          NaN    12.330196   NaN    NaN   
min         NaN             NaN          NaN    14.000000   NaN    NaN   
25%         NaN             NaN          NaN    32.000000   NaN    NaN   
50%         NaN             NaN          NaN    42.000000   NaN    NaN   
75%         NaN             NaN          NaN    52.000000   NaN    NaN   
max         NaN             NaN          NaN    87.000000   NaN    NaN   

       ResidenceCity ResidenceCounty ResidenceState DeathCity  \
count           4925            4303           3555      5092   
unique           370              84             26       221   
top         HARTFORD        HARTFORD             CT  HARTFORD   
freq             293            1203           3441       559   
mean             NaN             NaN            NaN       NaN   
std              NaN             NaN            NaN       NaN   
min              NaN             NaN            NaN       NaN   
25%              NaN             NaN            NaN       NaN   
50%              NaN             NaN            NaN       NaN   
75%              NaN             NaN            NaN       NaN   
max              NaN             NaN            NaN       NaN   

                   ...              Tramad Morphine_NotHeroin Hydromorphone  \
count              ...                 130                 42            25   
unique             ...                   1                  5             1   
top                ...                   Y                  Y             Y   
freq               ...                 130                 38            25   
mean               ...                 NaN                NaN           NaN   
std                ...                 NaN                NaN           NaN   
min                ...                 NaN                NaN           NaN   
25%                ...                 NaN                NaN           NaN   
50%                ...                 NaN                NaN           NaN   
75%                ...                 NaN                NaN           NaN   
max                ...                 NaN                NaN           NaN   

       Other OpiateNOS AnyOpioid MannerofDeath  \
count    431        88      2464          5095   
unique   132         1         2             5   
top      PCP         Y         Y      Accident   
freq      48        88      2459          5066   
mean     NaN       NaN       NaN           NaN   
std      NaN       NaN       NaN           NaN   
min      NaN       NaN       NaN           NaN   
25%      NaN       NaN       NaN           NaN   
50%      NaN       NaN       NaN           NaN   
75%      NaN       NaN       NaN           NaN   
max      NaN       NaN       NaN           NaN   

                                 DeathCityGeo  \
count                                    5095   
unique                                    373   
top     Hartford, CT\n(41.765775, -72.673356)   
freq                                      376   
mean                                      NaN   
std                                       NaN   
min                                       NaN   
25%                                       NaN   
50%                                       NaN   
75%                                       NaN   
max                                       NaN   

                             ResidenceCityGeo                InjuryCityGeo  
count                                    5002                         5017  
unique                                    289     

In [21]:
od = od.dropna(subset=['Date'])

In [31]:
od.columns

Index(['ID', 'Date', 'DateType', 'Age', 'Sex', 'Race', 'ResidenceCity',
       'ResidenceCounty', 'ResidenceState', 'DeathCity', 'DeathCounty',
       'Location', 'LocationifOther', 'DescriptionofInjury', 'InjuryPlace',
       'InjuryCity', 'InjuryCounty', 'InjuryState', 'COD', 'OtherSignifican',
       'Heroin', 'Cocaine', 'Fentanyl', 'FentanylAnalogue', 'Oxycodone',
       'Oxymorphone', 'Ethanol', 'Hydrocodone', 'Benzodiazepine', 'Methadone',
       'Amphet', 'Tramad', 'Morphine_NotHeroin', 'Hydromorphone', 'Other',
       'OpiateNOS', 'AnyOpioid', 'MannerofDeath', 'DeathCityGeo',
       'ResidenceCityGeo', 'InjuryCityGeo'],
      dtype='object')

We need to count specific drug deaths by converting "Y" in each column

In [33]:
def drug(value, column):
    if(value == 'Y'):
        return str(column).upper()
    else:
        return value

for column in od.loc[:, 'Heroin':'AnyOpioid']:
    if(column != 'Other'):
        od[column] = od[column].map(lambda value: drug(value, column))
        
od.loc[:, 'Heroin':'AnyOpioid'].head()

Heroin  Cocaine  Fentanyl FentanylAnalogue Oxycodone Oxymorphone Ethanol  \
0     NaN      NaN  FENTANYL              NaN       NaN         NaN     NaN   
1     NaN  COCAINE       NaN              NaN       NaN         NaN     NaN   
2  HEROIN  COCAINE       NaN              NaN       NaN         NaN     NaN   
3  HEROIN      NaN  FENTANYL              NaN       NaN         NaN     NaN   
4     NaN      NaN  FENTANYL              NaN       NaN         NaN     NaN   

   Hydrocodone  Benzodiazepine Methadone Amphet Tramad Morphine_NotHeroin  \
0  HYDROCODONE  BENZODIAZEPINE       NaN    NaN    NaN                NaN   
1          NaN             NaN       NaN    NaN    NaN                NaN   
2          NaN             NaN       NaN    NaN    NaN                NaN   
3          NaN             NaN       NaN    NaN    NaN                NaN   
4          NaN             NaN       NaN    NaN    NaN                NaN   

  Hydromorphone Other OpiateNOS  AnyOpioid  
0           NaN   NaN       NaN        NaN  
1           NaN   NaN       NaN        NaN  
2           NaN   NaN       NaN  ANYOPIOID  
3           NaN   NaN       NaN  ANYOPIOID  
4           NaN   NaN       NaN        NaN

In [34]:
#Create a new column and append the values in the drug columns to track multiple-use
od['drugs_used'] = od.loc[:, 'Heroin':'AnyOpioid'].apply(lambda value: ', '.join(value[value.notnull()]), axis = 1)
od['drugs_used'].value_counts()

HEROIN                                                                                 340
COCAINE                                                                                215
HEROIN, ANYOPIOID                                                                      209
FENTANYL, ANYOPIOID                                                                    200
HEROIN, FENTANYL, ANYOPIOID                                                            162
HEROIN, COCAINE                                                                        143
FENTANYL                                                                               107
HEROIN, FENTANYL                                                                       101
COCAINE, FENTANYL, ANYOPIOID                                                            99
HEROIN, ETHANOL                                                                         96
HEROIN, COCAINE, ANYOPIOID                                                              86